src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -611,7 +611,13 @@ def get_transitive_resolved_jars(my_coord, resolved_jars):
                 jars_per_target.append((t, jars_to_digest))
 
         for target, jars_to_add in self.add_directory_digests_for_jars(jars_per_target):
-            compile_classpath.add_jars_for_targets([target], conf, jars_to_add)
+            if override_classifiers is not None:
+                for jar in jars_to_add:
+                    compile_classpath.add_jars_for_targets(
+                        [target], jar.coordinate.classifier, [jar]
+                    )
+            else:
+                compile_classpath.add_jars_for_targets([target], conf, jars_to_add)
```

Comment:
> The issue is that coursier and ivy use different conf names for sources and javadocs. Namely coursier overloads the conf key src_doc for both javadocs and sources, which means they cannot coexist in the export-dep-as-jar json output as separate keys. We need the keys to be consistent between coursier and ivy to provide a stable interface so I figured I would fix it in the resolve.

Thanks for the context! Ivy resolve would be removed in the next release, would we be able to simplify the logic then? If so, could you add a TODO?

Reply:
I don't think we would be able to simplify this logic much, unless we want to switch over to the coursier conf names, which IMO aren't as clear as the ivy confs, especially for an interface with IDE where we want to be clear which jar is a sources jar vs javadocs jar. Coursier lumps them together.